test/test_torch.py

```diff
@@ -7036,11 +7031,13 @@ def test_pinverse(self, device, dtype):
         def run_test(M):
             # Testing against definition for pseudo-inverses
             MPI = torch.pinverse(M)
+            MPI_ = MPI.cpu().numpy()
+            M_ = M.cpu().numpy()
```

Comment:
would just M.cpu() and torch.matmul work to avoid tf32, or do you need np.matmul?

Reply:
Did't we decide to use numpy to do the computation everywhere?

Reply:
I don't know if we're that prescriptive (yet), but this should be fine. We definitely want to use NumPy for reference impls.